src/app/screens/HomeScreen/index.tsx

```diff
@@ -2,12 +2,15 @@ import React from 'react';
 import { FlatList, ListRenderItem } from 'react-native';
 import BookDetail from '@components/BookDetail';
 import { Book } from '@interfaces/book';
+import { Navigation } from '@interfaces/navigation';
+import { DETAILSCREEN } from '@constants/screenNames';
 
 import { DATA } from './constants';
 import styles from './styles';
 
-function HomeScreen() {
-  const renderBook: ListRenderItem<Book> = ({ item }) => <BookDetail book={item} />;
+function HomeScreen({ navigation }: Navigation) {
+  const onPress: () => void = () => navigation.navigate(DETAILSCREEN);
+  const renderBook: ListRenderItem<Book> = ({ item }) => <BookDetail book={item} onPress={onPress} />;
```

Comment:
I think the type `() => void` is not necessary. In some cases, the const inherits the type automatically
Also, use more descriptive names and the `handle/on` pattern for actions, for example
```
const handlePressBook = () => navigation.navigate(DETAIL_SCREEN);
const renderBook: ListRenderItem<Book> = ({ item }) => <BookDetail book={item} onPress={handlePressBook} />;
```